### PR TITLE
trunk for software quality

### PR DESCRIPTION
**Pull request by InnocentBug**
# Description
Added trunk for automatic software quality controlling

## Changes
Added the max line length to 128
## Known Issues
Accidentally based on #30 

All branches have to use trunk to adjust to the format changes.
## Notes

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -4,4 +4,8 @@
 
 ## Known Issues
 
-## Notes
\ No newline at end of file
+## Notes
+
+## Checklist
+
+- [ ] My name is on the list of contributors (`CONTRIBUTORS.md`) in the pull request source branch.
```
Comment (nh916) [inline .github/pull_request_template.md:11]: whats this?

just a checklist to be sure their name is included in the list of contributors?

**Review by nh916 — APPROVED** [review]

File: .github/pull_request_template.md (hunk 1 above)
Comment (InnocentBug) [reply .github/pull_request_template.md:11]: yes, just to make sure that we have everyone in the contributors list.

It reminds new people to add their name and us to make sure they are listed.

**Comment by InnocentBug** [comment]
Check out [this](https://docs.trunk.io/docs/check), if you would like to what this is about.

File: .github/pull_request_template.md (hunk 1 above)
Comment (nh916) [reply .github/pull_request_template.md:11]: thats a good idea!

I can dig it!

File: .github/workflows/trunk.yml [hunk 2]
```diff
@@ -0,0 +1,20 @@
+name: CI
+
+on:
+  push:
```
Comment (bearmit) [inline .github/workflows/trunk.yml:4]: It means there is no build after a merge. is it what you want?

**Review by bearmit — APPROVED** [review]
I don't know want to tell. Just a minor comment on the CI.

File: .github/workflows/trunk.yml (hunk 2 above)
Comment (InnocentBug) [reply .github/workflows/trunk.yml:4]: good catch. Thank you

**Review by bearmit — APPROVED** [review]

**Pull request merged**